agent/api/task/task.go

```diff
@@ -709,6 +734,92 @@ func (task *Task) getAllASMSecretRequirements() map[string]apicontainer.Secret {
 	return reqs
 }
 
+// hasLogRouter returns true if the task has a log router container.
+func (task *Task) hasLogRouter() bool {
+	for _, container := range task.Containers {
+		if container.LogRouter != nil { // This is a log router container.
+			return true
+		}
+	}
+	return false
+}
+
+// initializeLogRouterResource initializes the log router task resource and adds it as a dependency of the
+// log router container.
+func (task *Task) initializeLogRouterResource(config *config.Config, resourceFields *taskresource.ResourceFields) error {
+	containerToLogOptions, err := task.collectLogRouterOptions()
+	if err != nil {
+		return errors.Wrap(err, "unable to initialize log router resource")
+	}
+
+	var logRouterResource *logrouter.LogRouterResource
+	for _, container := range task.Containers {
+		logRouter := container.LogRouter
+		if logRouter != nil {
+			var ec2InstanceID string
+			if container.Environment != nil && container.Environment[awsExecutionEnvKey] == ec2ExecutionEnv {
```

Comment:
why is this check needed?

Reply:
discussed offline. we don't want to show the instance id for all the cases